R/Figures.R

```diff
@@ -864,30 +864,62 @@ TADA_TwoCharacteristicScatterplot <- function(.data, id_cols = "TADA.ComparableD
   }
 
   if (!"TADA.ComparableDataIdentifier" %in% id_cols) {
-    print("Note: TADA.ComparableDataIdentifier not found in id_cols argument and is highly recommended.")
+    stop("Note: TADA.ComparableDataIdentifier not found in id_cols argument.")
   }
 
   # check that groups are in id_cols
   id <- unlist(unique(.data[, id_cols]))
   if (any(!groups %in% id)) {
     stop("The 'groups' vector contains one or more inputs that are not found within your input dataset. Check spelling and try again.")
   }
+  
+  # check that groups are length two for comparing two characteristics
+  if (length(groups) != 2) {
+    stop("The 'groups' vector does not contain two characteristics to compare.")
+  }
 
+  # if another grouping column wants to be plotted, check that groups2 are in id_cols2
+  
+  if (!is.null(id_cols2)) {
+    id2 <- unlist(unique(.data[, id_cols2]))
+    if (any(!groups2 %in% id2)) {
+    stop("The 'groups2' vector contains one or more inputs that are not found within your input dataset. Check spelling and try again.")
+    }
+  }
+  
+  # check that groups2 are length two for comparing two groups
```

Comment:
Should the function generate a plot anyway if there is only one group? There wouldn't be a comparison, but it seems like a plot could still be generated? Or should the message prompt the user to use TADA_Scatterplot?